src/reducers/scrollState.js

```diff
@@ -72,14 +71,18 @@ function scrollState(state = DEFAULT_STATE, action) {
 
       if (newProps.scrollTop && newProps.scrollTop !== oldProps.scrollTop) {
         state = scrollTo(state, newProps.scrollTop);
-        state = updateVisibleRows(state);
       }
 
       if (newProps.scrollToRow && newProps.scrollToRow !== oldProps.scrollToRow) {
         state = scrollToRow(state, newProps.scrollToRow);
-        state = updateVisibleRows(state);
       }
 
+      if (newProps.rowsCount !== state.rowsCount) {
+        state = updateRowCount(state, newProps);
+      }
+
+      state = updateVisibleRows(state);
```

Comment:
Could we create a needsVisibleRowsUpdate boolean local var and init to false.  Then set it to true in each of the 3 if statements.  Then we can wrap this call in an if statment on that boolean.  Thoughts?